### PR TITLE
revert module config

### LINKED ISSUE #7088: issue: react hook form breaks blitz.js/next.js build since 7.20
### Version Number

7.20.1

### Codesandbox/Expo snack

https://github.com/JohannesKlauss/blitzjs-rhf-build-error

### Steps to reproduce

1. Check out the repo mentioned above
2. Run `yarn install`
3. Run `yarn run build`

### Expected behaviour

The build completes without error.

I guess this is related to this commit: https://github.com/react-hook-form/react-hook-form/commit/461c7db172f34e28c2439f64b23dfd173191a58d

Not quite sure though. I can confirm that is worked without a problem for version `7.19.5`. Let me know if you need any help!

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

```shell
(node:6594) UnhandledPromiseRejectionWarning: Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js
require() of ES modules is not supported.
require() of /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.cjs.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/package.json.

    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1080:13)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js:1:7)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:6594) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:6594) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:6594) UnhandledPromiseRejectionWarning: Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js
require() of ES modules is not supported.
require() of /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.cjs.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/package.json.

    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1080:13)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js:1:7)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
(node:6594) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)

> Build error occurred
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js
require() of ES modules is not supported.
require() of /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.cjs.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/package.json.

    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1080:13)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js:1:7)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14) {
  type: 'NodeError',
  code: 'ERR_REQUIRE_ESM'
}
error Command failed with exit code 1.
```
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bluebill1049**
close #7088

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -136,6 +135,6 @@
     "url": "https://opencollective.com/react-hook-form"
   },
   "engines": {
-    "node": ">=14.16"
+    "node": ">=12.22"
```
Comment (barrymay) [inline package.json:138]: you could just revert module to and keep the rest.  That said, node 12 is in maintenance mode until April 2022 to technically we may not want to require 14 until then.

Comment (bluebill1049) [reply package.json:138]: thanks @barrymay i have referred to this PR: https://github.com/react-hook-form/react-hook-form/pull/7039

Comment (bluebill1049) [reply package.json:138]: i will change to 12 actually.

File: package.json [hunk 2]
```diff
@@ -136,6 +135,6 @@
     "url": "https://opencollective.com/react-hook-form"
   },
   "engines": {
-    "node": ">=14.16"
+    "node": ">=12.0"
```
Comment (barrymay) [inline package.json:138]: I made a comment elsewhere but I can't find it - I'm fine with this revert (maybe test could stay as it was at most).  If we support 12.* I think it's fine for now - technically node has 12 in maintenance mode until April 2022, so unless there's a significant benefit to the upgrade I say we keep it in relatively up-to-date 12.*

**Review by barrymay — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (bluebill1049) [reply package.json:138]: 👍  thanks.

File: package.json [hunk 3]
```diff
@@ -2,7 +2,6 @@
   "name": "react-hook-form",
   "description": "Performant, flexible and extensible forms library for React Hooks",
   "version": "7.20.1",
-  "type": "module",
   "main": "dist/index.cjs.js",
```
Comment (yordis) [inline package.json:5]: Based on these two lines combined, the `type` must be `commonjs`

The ultimate fix is to configure correctly `exports` if I am not mistaken: https://nodejs.org/api/packages.html#package-entry-points

```json
{
  "main": "./main.js",
  "exports": {
    "node": {
      "import": "./feature-node.mjs",
      "require": "./feature-node.cjs"
    },
    "default": "./feature.mjs",
  }
}
```

Comment (bluebill1049) [reply package.json:5]: thanks @yordis would you like to send us a PR?

Comment (yordis) [reply package.json:5]: @bluebill1049 aye! let me see if I can make it work

Comment (bluebill1049) [reply package.json:5]: thanks 👍

**Pull request merged**